parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+                notices_others_last.append(notice['displayName'])
+            elif notice_short in SPECIAL_NOTICES:
+                notices_special.append(notice['displayName'])
+            else:
+                if sub_notices:
+                    notices_others.append(notice['displayName'])
+                else:
+                    notices_others.append('mit %s' % notice['displayName'])
+
+    if sub_notices:
+        if len(notices_allergens) > 0:
+            notices_allergens[0] = 'enthält %s' % notices_allergens[0]
+        if len(notices_allergens) > 1:
+            notices_allergens[-2] = (
+                '%s und %s' % (notices_allergens[-2], notices_allergens[-1]))
+            del notices_allergens[-1]
+        if len(notices_others) > 0:
+            notices_others[0] = 'mit %s' % notices_others[0]
+        if len(notices_others) > 1:
+            notices_others[-2] = (
+                '%s und %s' % (notices_others[-2], notices_others[-1]))
+            del notices_others[-1]
+
+    return (notices_special + notices_allergens +
+            notices_others + notices_others_last)
+
+
+def build_notes(notices, components):
+    global base_data
+
+    components_all = []
+
+    notices_list = get_notices(notices)
+
+    for component in components:
+        component_string = 'dazu ' + component['name']
+        if len(component['notices']) > 0:
+            component_notices_list = get_notices(component['notices'], True)
+            component_string += ' (%s)' % ', '.join(component_notices_list)
+        components_all.append(component_string)
+
+    return components_all + notices_list
+
+
+def build_hours(opening_hours):
+    if not opening_hours:
+        return []
+
+    start = datetime.datetime.strptime(
+        opening_hours['start'], UTC_DATE_STRING).time()
+    end = datetime.datetime.strptime(
+        opening_hours['end'], UTC_DATE_STRING).time()
+    return [
+        'verfügbar von %s bis %s Uhr' %
+        (datetime.time.strftime(start, "%H:%M"),
+            datetime.time.strftime(end, "%H:%M"))
+    ]
+
+
+def build_location(description):
+    return ['Ort: %s' % description] if description else []
+
+
+def parse_url(url, today=False):
+    global base_data
+
+    canteen = LazyBuilder()
+    with urlopen(url) as response:
+        data = json.loads(response.read().decode())
+
+    for day in data['days']:
+        date = day.get('date')
+
+        if today and (datetime.datetime.now().date() !=
+                      datetime.datetime.strptime(date, UTC_DATE_STRING).date()):
+            continue
+
+        for counter in day['counters']:
+            counter_name = counter['displayName']
+            counter_description = counter['description']
+            counter_hours = counter.get('openingHours')
+            counter_color = counter.get('color')
+
+            for meal in counter['meals']:
+                if 'knownMealId' in meal:
+                    print('knownMealId: %s' % meal['knownMealId'])
```

Comment:
There is currently one known meal in the base data:
```json
"knownMeals": {
  "gebfl": {
    "displayName": "Gebackener Fleischkäse"
  }
}
```

Should this be used somehow? Do these meals maybe have no displayName, which would cause the `counter['displayName']` above to fail?

Reply:
This is meant to allow recognizing recurring meals, for features like marking meals as favorites. (Simply matching the title isn't enough because there often are minor differences.)

So far, I couldn't actually convince Studentenwerk Saarland to build the list of recurring meals, so this list only contains this single entry as an example.